src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -0,0 +1,146 @@
+      if (ip) nodeSet.add(ip);
+    }
+  } catch (err) {
+    log.error("error fetching node IPs:", err);
+  }
+}
+
+/**
+ * Returns the egress CIDRs of all known nodes as network policy peers.
+ * If none are known, defaults to 0.0.0.0/0 and logs a warning.
+ */
+export function nodeCIDRs(): V1NetworkPolicyPeer[] {
+  const policies = buildNodePolicies([...nodeSet]);
+  if (policies.length > 0) return policies;
+
+  log.warn("Unable to get API server CIDR, defaulting to 0.0.0.0/0");
+  return [anywhere];
+}
+
+/**
+ * When a node is created or updated, if it's Ready, add its IP to the set,
+ * rebuild the policies, and update the NetworkPolicies.
+ */
+export async function updateKubeNodesFromCreateUpdate(node: kind.Node) {
+  const isReady = node.status?.conditions?.some(
+    (condition: V1NodeCondition) => condition.type === "Ready" && condition.status === "True",
+  );
+
+  if (isReady) {
+    const ip = getNodeInternalIP(node);
+    if (ip) nodeSet.add(ip);
+  }
+
+  const policies = buildNodePolicies([...nodeSet]);
+  await updateKubeNodesNetworkPolicies(policies);
+}
+
+/**
+ * When a node is deleted, remove its IP from the set, rebuild the policies,
+ * and update the NetworkPolicies.
+ */
+export async function updateKubeNodesFromDelete(node: kind.Node) {
+  const ip = getNodeInternalIP(node);
+  if (ip) nodeSet.delete(ip);
+
+  const policies = buildNodePolicies([...nodeSet]);
+  await updateKubeNodesNetworkPolicies(policies);
+}
+
+/**
+ * Fetch all Kubernetes nodes.
+ */
+async function fetchKubernetesNodes(): Promise<KubernetesListObject<kind.Node>> {
+  return K8s(kind.Node).Get();
+}
+
+/**
+ * Update all NetworkPolicies labeled with uds/generated=KubeNodes to
+ * reflect the given node CIDRs.
+ */
+export async function updateKubeNodesNetworkPolicies(newNodes: V1NetworkPolicyPeer[]) {
+  const netPols = await K8s(kind.NetworkPolicy)
+    .WithLabel("uds/generated", RemoteGenerated.KubeNodes)
+    .Get();
+
+  for (const netPol of netPols.items) {
+    if (!netPol.spec) {
+      log.warn(
+        `KubeNodes NetworkPolicy ${netPol.metadata?.namespace}/${netPol.metadata?.name} is missing spec.`,
+      );
+      continue;
+    }
+
+    if (netPol.spec.egress) {
+      netPol.spec.egress[0] = netPol.spec.egress[0] || { to: [] };
+      netPol.spec.egress[0].to = newNodes;
+    }
+
+    if (netPol.metadata) {
+      // Remove managed fields to prevent server-side apply errors
+      netPol.metadata.managedFields = undefined;
+    }
+
+    log.debug(
+      `Updating KubeNodes NetworkPolicy ${netPol.metadata?.namespace}/${netPol.metadata?.name} with new CIDRs.`,
+    );
+
+    try {
+      await K8s(kind.NetworkPolicy).Apply(netPol, { force: true });
+    } catch (err) {
+      let message = err.data?.message || "Unknown error while applying KubeNode network policies";
+      if (UDSConfig.kubeApiCidr) {
+        message +=
+          ", ensure that the KUBEAPI_CIDR override configured for the operator is correct.";
+      }
```

Comment:
This error message addition is irrelevant in the context of the `KubeNodes` policies, although it does bring up a potential question around this - we may want to add a manual "override" similar to the `UDSConfig.kubeApiCidr` override to allow people to opt out of auto-updates. The usage/explanation of that is documented [here](https://uds.defenseunicorns.com/reference/configuration/uds-networking-configuration/). It doesn't require too many code changes to support if we wanted to add it (should be able to copy most of what was done for KubeAPI).

Reply:
(If we do end up adding this manual config override let's make sure we also add to that same doc)

Reply:
Support for this was added in 15e6235d , I believe